### ISSUE TITLE
[BUG] ForecastingPipeline.update fails with ValueError on exogeneous data

### ISSUE THREAD

**Issue by jamontol** [alan-turing-institute/sktime#1168]
**Describe the bug**

After fit and predict with the forecaster model correctly, an error is raised when trying to update forecaster model with new exogenous X data

```
Exception has occurred: ValueError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
Must specify axis=0 or 1
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\pandas\core\generic.py", line 8966, in _align_series
    raise ValueError("Must specify axis=0 or 1")
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\pandas\core\generic.py", line 8834, in align
    fill_axis=fill_axis,
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\pandas\core\frame.py", line 4112, in align
    broadcast_axis=broadcast_axis,
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\pandas\core\frame.py", line 6331, in combine
    this, other = self.align(other, copy=False)
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\pandas\core\frame.py", line 6448, in combine_first
    return self.combine(other, combiner, overwrite=False)
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\sktime\forecasting\base\_base.py", line 439, in _update_y_X
    self._X = X.combine_first(self._X)
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\sktime\forecasting\base\_base.py", line 262, in update
    self._update_y_X(y, X)
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\sktime\forecasting\compose\_pipeline.py", line 240, in _update
    forecaster.update(y=y, X=X, update_params=update_params)
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\sktime\forecasting\base\_base.py", line 264, in update
    self._update(y=y, X=X, update_params=update_params)
  File "C:\Users\jrml\OneDrive - GFT Technologies SE\Documents\SPHINX\zf-electricity-demand-forecast\SPRINT_2\API-SPHINX\prediction-cloud-sktime\app\Forecasting\forecaster.py", line 187, in predict
    forecaster_model.update(y=y, X=X, update_params=False)
```

**To Reproduce**

y is a Series object with 1 row and X is a DataFrame object with 1 row and n columns (n=1 exogenous variable) with  DatetimeIndex index updated to a new period (15 min) after training

```python
strategy='direct'
window_length=4
regressor = RandomForestRegressor(**params)

forecaster_model = ForecastingPipeline(steps=[
            ("scaler_X", TabularToSeriesAdaptor(MinMaxScaler())), # transform X exogenous
            ("forecaster", TransformedTargetForecaster( # y
                [
                    ("scaler", OptionalPassthrough(TabularToSeriesAdaptor(MinMaxScaler()))),
                    ("forecast", make_reduction(regressor, window_length=window_length, strategy=strategy))
                ])
            )
            ])

forecaster_model.fit(target, features, fh = self.fh)
df_y_pred = forecaster_model.predict(fh=self.fh, X=X)  # X is ignored here
forecaster_model.update(y=y, X=X, update_params=False)

```
I've also  tried update_params=True but it raises the same error

**Expected behavior**

The model should be updated with no errors

**Additional context**

The error occurs when forecaster_model.update() invokes update() method in ```BaseForecaster``` parent class and this invokes self._update() method on ```ForecastingPipeline``` child class. Inside this method, update() in ```BaseForecaster``` parent class is invoked again from forecaster.update() 

**Versions**
v0.7.0

**Comment by aiwalter** [comment]
Thx, I will look into it

**Comment by aiwalter** [comment]
@jamontol could you maybe provide a reproducable example with the `y,X=load_longley()` data and which `fh` you have defined...?

**Comment by jamontol** [comment]
Hi, I tested with load_longley() and the update method apparently works, it doesn't raise any error, but now the second prediction raises the following error:

``` 
Input contains NaN, infinity or a value too large for dtype('float32').
  File "C:\Users\jrml\AppData\Local\Continuum\anaconda3\envs\time-series\Lib\site-packages\sklearn\utils\validation.py", line 106, in _assert_all_finite
    msg_dtype if msg_dtype is not None else X.dtype)
....
```

```
y, X = load_longley()
fh= np.arange(1,4)
...
forecaster_model.fit(y[:-2], X[:-2])
df_y_pred = forecaster_model.predict(fh=self.fh, X=X[-2:-1])  # X is ignored here
forecaster_model.update(y[-2:-1], X[-2:-1], update_params=False)
df_y_pred_2 = forecaster_model.predict(fh=self.fh, X=X[-1:]) # X is ignored here
```

The only difference with my dataset is the index type, my index is a <class 'pandas.core.indexes.datetimes.DatetimeIndex'>
and load_longley index is a <class 'pandas.core.indexes.period.PeriodIndex'>

**Comment by aiwalter** [comment]
Could you paste the full error pls?

**Comment by jamontol** [comment]
OK, 

it works, the problem with my dataset was that it only had 1 exogenous variable (temperature) in form of a DataFrame that I created with 
X=f_data.drop(['energy'], axis=1) dropping the y column (energy)
When a create X  as a Series with X = t_data_period['temperature'] everything works fine. 

You can close this issue.

Sorry for the waste of time.

**Comment by aiwalter** [comment]
No worries
